backends/tfhe-cuda-backend/cuda/include/integer/compression/compression_utilities.h

```diff
@@ -118,9 +127,10 @@ template <typename Torus> struct int_decompression {
                                        gpu_indexes[0], gpu_memory_allocated);
     cuda_drop_with_size_tracking_async(tmp_indexes_array, streams[0],
                                        gpu_indexes[0], gpu_memory_allocated);
-
-    decompression_rescale_lut->release(streams, gpu_indexes, gpu_count);
-    delete decompression_rescale_lut;
+    if constexpr (std::is_same_v<Torus, uint64_t>) {
+      decompression_rescale_lut->release(streams, gpu_indexes, gpu_count);
+      delete decompression_rescale_lut;
```

Comment:
I think in the new pattern we should also do
decompression_rescale_lut = nullptr;